subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/EclipseClasspath.java

```diff
@@ -161,10 +161,15 @@
 
     private final Property<Boolean> containsTestFixtures;
 
+    private final Property<String> testSourceSetNamePattern;
+    private final Property<String> testDependencyConfigurationNamePattern;
+
     @Inject
     public EclipseClasspath(org.gradle.api.Project project) {
         this.project = project;
         this.containsTestFixtures = project.getObjects().property(Boolean.class).convention(false);
+        this.testSourceSetNamePattern = project.getObjects().property(String.class).convention(".*[tT][eE][sS][tT].*");
```

Comment:
I think the danger in testSuites is that the DSL might encourage people naming their suites just `integration` or `functional` and leaving off the `Test` suffix, so they would not get discovered here.

Reply:
I think probably a combination of both your change to pickup custom SourceSets which are configured outside of test suites, and reading the suites directly per my suggestion below (or something similar) will be needed to get everything which ought to be marked 'test'.